activerecord/lib/active_record/store.rb

```diff
@@ -225,10 +225,7 @@ def self.read(object, attribute, key)
 
         def self.write(object, attribute, key, value)
           prepare(object, attribute)
-          if value != read(object, attribute, key)
```

Comment:
Still thinking about it, but should we bring the `!=` check back?
It won't change the behavior for the script I mentioned in the comments but it will still solve the issue as the test we added in this PR still passes after re-adding `!=`  check

Reply:
Can you open a PR with an extra test? We can then iterate on it.